website/static/js/tests/contributors.test.js

```diff
@@ -0,0 +1,196 @@
+           var hardReset = () => {
+               var addContribs = new addContributors('nothing', 'Fake title', '12345', 'parent', 'Parent title');
+               vm = addContribs.viewModel;
+           };
+           before(() => {
+               hardReset();
+           });
+
+           describe('getContributors', () => {
+               var shouldBe = ['a1234', 'b1234', 'c1234'];
+
+               it('should be a list of ids', () => {
+                   vm.getContributors()
+                       .always(() => {
+                           assert.equal(vm.contributors(), shouldBe);
+                       });
+               });
+           });
+
+           describe('getEditableChildren', () => {
+               var shouldBe = [
+                   {
+                       id: '23456',
+                       indent: 0,
+                       margin: '25px'
+                   },
+                   {
+                       id: '23457',
+                       indent: 1,
+                       margin: '50px'
+                   },
+                   {
+                       id: '23458',
+                       indent: 2,
+                       margin: '75px'
+                   }
+               ];
+
+               it('should be a list of indented nodes (25px offset)', () => {
+                   vm.getEditableChildren()
+                       .always(() => {
+                           assert.equal(vm.nodes(), shouldBe);
+                       });
+               });
+           });
+
+           describe('Visible', () => {
+               describe('addAllVisible', () => {
+                   it('should be false', () => {
+                       vm.contributors([
+                           'a1234'
+                       ]);
+                       vm.results([
+                           {
+                               id: 'a1234'
+                           },
+                           {
+                               id: 'b1234'
+                           },
+                           {
+                               id: 'c1234'
+                           }
+                       ]);
+                       vm.selection([
+                           {
+                               id: 'b1234'
+                           },
+                           {
+                               id: 'c1234'
+                           }
+                       ]);
+                       assert.isFalse(vm.addAllVisible());
+                   });
+
+                   it('should be true', (done) => {
+                       vm.selection([
+                           {
+                               id: 'b1234'
+                           }
+                       ]);
+                       assert.isTrue(vm.addAllVisible());
+                       done();
+                   });
+               });
+
+               describe('removeAllVisible', () => {
+                   it('should be true', () => {
```

Comment:
The nice thing about this testing structure is the power in your descriptions e.g. 

    describe('removeAllVisible', () => {
        it('should return true after removing <whatever>', () => { ...